### PR TITLE
Add opensearch to site

### LINKED ISSUE #1121: Add opensearch
It would be really nice to have the option to search for charms directly from the address bar like so: `ch nginx`

Screenshot from 2021-09-08 16-50-39

### PR DESCRIPTION
**Pull request by steverydz**
## Done
Added [opensearch](https://developer.mozilla.org/en-US/docs/Web/OpenSearch)

## QA
Can't QA until on live

## Issue
Fixes #1121

### CODE REVIEW

File: static/opensearch.xml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<OpenSearchDescription xmlns="http://a9.com/-/spec/opensearch/1.1/">
+	<ShortName>Charmhub</ShortName>
+	<Description>Charmhub - The Open Operator Collection</Description>
+	<InputEncoding>[UTF-8]</InputEncoding>
```
Comment (mrgnr) [inline static/opensearch.xml:4]: I think the square brackets are not needed here and in the next line.

**Review by mrgnr — CHANGES_REQUESTED** [review]
Aside from the square brackets, the code looks good to me from the documentation I've read. It might be good to have another pair of :eyes: before merging though.

File: static/opensearch.xml (hunk 1 above)
Comment (steverydz) [reply static/opensearch.xml]: Done

**Comment by steverydz** [comment]
@mrgnr I've made the requested change. Can you please approve if you're happy?

File: static/opensearch.xml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+<OpenSearchDescription xmlns="http://a9.com/-/spec/opensearch/1.1/">
+	<ShortName>Charmhub</ShortName>
+	<Description>Charmhub - The Open Operator Collection</Description>
+	<InputEncoding>UTF-8</InputEncoding>
+	<Image width="16" height="16" type="image/x-icon">[https://assets.ubuntu.com/v1/5d4edefd-jaas-favicon.png]</Image>
```
Comment (anthonydillon) [inline static/opensearch.xml:5]: Does this image need to be 16x16 as its 256 at the moment. Also the square brackets here will fail?

**Review by anthonydillon — CHANGES_REQUESTED** [review]
Looks good (interesting I didnt know know about this). Few comments about the image used.

File: static/opensearch.xml (hunk 2 above)
Comment (anthonydillon) [reply static/opensearch.xml:5]: Square brackets are used in the example to show something to replace.

Comment (anthonydillon) [reply static/opensearch.xml:5]: Type should be "image/png" to. x-icon is for `.ico` images

Comment (steverydz) [reply static/opensearch.xml]: @anthonydillon I've replaced the image with 64x64 so that should make it look good on higher res screens too

**Review by mrgnr — APPROVED** [review]
LGTM :+1:

**Pull request merged**